src/clj/rems/db/user_settings.clj

```diff
@@ -25,42 +26,41 @@
       validate-user-settings
       json/generate-string))
 
-(s/defschema PartialUserSettings
-  (map-keys s/optional-key UserSettings))
-
-(def ^:private coerce-partial-user-settings
-  (coerce/coercer! PartialUserSettings json/coercion-matcher))
+(def ^:private coerce-user-settings
+  (coerce/coercer! UserSettings json/coercion-matcher))
 
 (defn- json->settings [json]
-  ;; Allows missing keys, so we don't need to write migrations
-  ;; if we add new keys to user settings. Migrations are needed if
-  ;; we remove keys or make their validation stricter.
   (when json
     (-> json
         json/parse-string
-        coerce-partial-user-settings)))
+        coerce-user-settings)))
 
 (defn get-user-settings [user]
   (merge (default-settings)
          (when user
            (json->settings (:settings (db/get-user-settings {:user user}))))))
 
-(defn validate-new-settings [{:keys [language notification-email] :as settings}]
-  (into {} [(when (contains? (set (:languages env)) language)
-              {:language language})
-            (when (and notification-email (re-matches +email-regex+ notification-email))
-              {:notification-email notification-email})
-            (when (and (contains? settings :notification-email)
-                       (str/blank? notification-email))
-              ;; clear notification email to use identity provider's email instead
-              {:notification-email nil})]))
+(defn validate-new-settings [{:keys [language notification-email] :as new-settings}]
+  (let [valid-new-settings (merge (when (contains? (set (:languages env)) language)
+                                    {:language language})
+                                  (when (and notification-email (re-matches +email-regex+ notification-email))
+                                    {:notification-email notification-email})
+                                  (when (and (contains? new-settings :notification-email)
+                                             (str/blank? notification-email))
+                                    ;; clear notification email to use identity provider's email instead
+                                    {:notification-email nil})
+                                  (when-let [ega (:ega new-settings)]
```

Comment:
should we check the `:enable-ega` config here?

Reply:
Yes, this was made before that setting so it makes sense to add that check here too.